### PR TITLE
Add Exponentially scaled modified Bessel Op

### LINKED ISSUE #542: ENH: Implement exponentially scaled modified Bessel function Op
### Before

```python
pt.iv(J, a) / pt.exp(a)
```

### After

```python
pt.ive(J, a)
```

### Context for the issue:

Periodic HSGPs (now in PyMC 🥳 ) require `iv(J, a) / exp(a)` as normalizing factor. Using this directly could lead to overflows however (as `iv` grows like `exp(x)`). Instead `ive(nu,x) = iv(nu,x) / exp(abs(x))` exists, and as a bonus cheaper to compute than the combined!

After #403 we already have an implementation of `ive` in JAX, and it's in [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.ive.html), so should be straightforward!

### PR DESCRIPTION
**Pull request by dehorsley**
Fixes #542

As with `Iv`, uses scipy implementation in C code and tensorflow probability implementation for jax.

## Type of change

- [x] New feature / enhancement

### CODE REVIEW

**Comment by ricardoV94** [comment]
Looks great, could also be a nice touch to add a rewrite to convert the iv/exp form into ive so users writing the old form get the more stable one now automatically

**Comment by ricardoV94** [comment]
Could use PatternRewriter like we did here: https://github.com/pymc-devs/pytensor/pull/505

**Comment by dehorsley** [comment]
> Looks great, could also be a nice touch to add a rewrite to convert the iv/exp form into ive so users writing the old form get the more stable one now automatically

Sounds good. The tricky thing is that we need an `abs` in there unless we can prove the argument is positive and real. By which I mean the rewrite is:

`iv(nu, x)/exp(abs(x))` -> `ive(nu, x)`

but the use in PyMC is:

```python
a = 1 / pt.square(self.ls)
c = pt.where(J > 0, 2, 1)

q2 = c * pt.iv(J, a) / pt.exp(a)
```

So unless we carry some "positive real" tag(s) around, we won't pickup this kind of case. Maybe not worth the complexity, and just add the above rewrite?

Also, do you have a suggestion on how to handle the two forms? I.e. can I catch both these as one rewrite:

`iv(nu, x)/exp(abs(x))` -> `ive(nu, x)`
`iv(nu, x)*exp(-abs(x))` -> `ive(nu, x)`

Is there a canonicalizing rewrite that will handle that?

**Comment by ricardoV94** [comment]
Canonicalize wouldn't do that kind of modification. You can write a more typical rewrite with a function that handles both cases.

But since it needs to be positive maybe not worth adding for now. 

The kind of machinery in https://github.com/pymc-devs/pytensor/pull/525 would make it really easy to complement rewrites with positivity/negative analysis. This case may be a nice motivating example.

**Review by ricardoV94 — APPROVED** [review]

**Comment by dehorsley** [comment]
> But since it needs to be positive maybe not worth adding for now.

So happy to go as is, or would you like me to add the rewrite with the `abs` in there?
